rust/protocol/src/sealed_sender.rs

```diff
@@ -704,6 +803,58 @@ pub async fn sealed_sender_encrypt<R: Rng + CryptoRng>(
     sealed_sender_encrypt_from_usmc(destination, &usmc, identity_store, ctx, rng).await
 }
 
+/// This method implements a single-key single-recipient [KEM], described in [this Signal blog post]
+/// on the sealed sender mechanism.
+///
+/// [KEM]: https://en.wikipedia.org/wiki/Key_encapsulation
+/// [this Signal blog post]: https://signal.org/blog/sealed-sender/
+///
+/// ### Contrast with Multi-Recipient KEM
+/// The *single-recipient* KEM scheme implemented by this method partially derives the encryption
+/// key from the recipient's identity key, which would then require re-encrypting the same message
+/// multiple times to send to multiple recipients. In contrast,
+/// [sealed_sender_multi_recipient_encrypt] uses a *multi-recipient* KEM scheme which avoids this
+/// repeated work. However, that particular multi-recipient KEM also has a few caveats which may
+/// make it unwieldy for certain scenarios:
+/// 1. it requires a [SessionRecord] to exist already for the recipient, i.e. that a Double Ratchet
+///    message chain has previously been established in the [SessionStore] via
+///    [crate::process_prekey_bundle] after an initial [crate::PreKeySignalMessage] is recieved.
+/// 2. it ferries a lot of additional information in its encoding which makes the resulting message
+///    bulkier than the message produced by this method.
+/// 3. it does not currently correspond to a specific protobuf schema.
+///
+/// ### Overview
+/// The KEM scheme implemented by this method is described in [this Signal blog post]. The
+/// high-level steps of this process are listed below:
+/// 1. Generate a random key pair.
+/// 2. Derive a symmetric chain key, cipher key, and MAC key from a salt composed of the recipient's
+///    public key and the sender's public and private key pair.
+/// 3. Symmetrically encrypt the sender's public key using the cipher key and MAC key from (2) with
+///    AES-256 in CTR mode.
+/// 4. Derive a second symmetric cipher key and MAC key from a salt composed of the sender's private
+///    key, the recipient's public key, and the chain key from (2).
+/// 5. Symmetrically encrypt the underlying [UnidentifiedSenderMessageContent] using the cipher key
+///    and MAC key from (4) with AES-256 in CTR mode.
+/// 6. Send the ephemeral public key from (1) and the encrypted public key from (3) to the
+///    recipient, along with the encrypted message (5).
+///
+/// #### Wire Format
+/// The output of this method is encoded as a protobuf with an additional [u8] prepended to the
+/// output indicating that this message was sent using the single-recipient KEM scheme.
```

Comment:
I think it's just "indicating the version of Sealed Sender in use".

Reply:
Thanks! Changed.